src/components/CodeEditor.tsx

```
@@ -11,24 +11,24 @@ import { recordAttendance } from '@api/user';
 
 interface CodeEditorProps {
   code: string;
-  onChange: (newCode: string) => void;
-  onSubmit: () => Promise<boolean | undefined>;
+  onChange: (code: string) => void;
   onLanguageChange: (mode: string) => void;
+  onSubmit: () => Promise<void | undefined>;
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)